File: app/src/main/java/com/dpis/module/appconfig/presentation/AppConfigSheetInteractions.java
package com.dpis.module.appconfig.presentation;

import com.dpis.module.applist.AppListItem;

import android.view.View;
import com.dpis.module.appconfig.presentation.AppConfigDialogBinder;

public final class AppConfigSheetInteractions {
    private final AppConfigSheetModeValidationBinder modeValidationBinder;
    private final AppConfigSheetActionBinder actionBinder;

    public AppConfigSheetInteractions(AppConfigDialogBinder binder, AppConfigDialogBinder.Host host) {
        this.modeValidationBinder = new AppConfigSheetModeValidationBinder(binder, host);
        this.actionBinder = new AppConfigSheetActionBinder(binder, host);
    }

    public void bind(View dialogView,
            AppListItem item,
            AppConfigDialogBinder.AppConfigDialogViews views,
            AppConfigDialogBinder.AppConfigDialogState state,
            AppConfigDialogBinder.AppConfigDialogActionStyle style,
            boolean systemHooksEnabled) {
        modeValidationBinder.bindDialogValidation(dialogView, item, views, state, style, systemHooksEnabled);
        actionBinder.bindDialogActions(dialogView, item, views, state, style, systemHooksEnabled);
        modeValidationBinder.bindModeToggles(dialogView, item, views, state, style, systemHooksEnabled);
        actionBinder.bindTypefaceSelectorAction(dialogView, item, views, state, style, systemHooksEnabled);
    }
}

File: app/src/main/java/com/dpis/module/appconfig/presentation/AppConfigSheetInteractions.kt
package com.dpis.module.appconfig.presentation

import android.view.View
import com.dpis.module.applist.AppListItem

class AppConfigSheetInteractions(
    binder: AppConfigDialogBinder,
    host: AppConfigDialogBinder.Host,
) {
    private val modeValidationBinder = AppConfigSheetModeValidationBinder(binder, host)
    private val actionBinder = AppConfigSheetActionBinder(binder, host)

    fun bind(
        dialogView: View,
        item: AppListItem,
        views: AppConfigDialogBinder.AppConfigDialogViews,
        state: AppConfigDialogBinder.AppConfigDialogState,
        style: AppConfigDialogBinder.AppConfigDialogActionStyle,
        systemHooksEnabled: Boolean,
    ) {
        modeValidationBinder.bindDialogValidation(dialogView, item, views, state, style, systemHooksEnabled)
        actionBinder.bindDialogActions(dialogView, item, views, state, style, systemHooksEnabled)
        modeValidationBinder.bindModeToggles(dialogView, item, views, state, style, systemHooksEnabled)
        actionBinder.bindTypefaceSelectorAction(dialogView, item, views, state, style, systemHooksEnabled)
    }
}
